user/api_test.go

```diff
@@ -660,6 +660,32 @@ func Test_UpdateUser_Error_UnauthorizedRoles_User(t *testing.T) {
 	T_ExpectErrorResponse(t, response, 401, "Not authorized for requested operation")
```

Comment:
We could add the test patient -> caregiver role change in order to cover all possible changes.